tests/analyzer/test_data_diagnosis.py

```diff
@@ -176,3 +187,35 @@ def test_data_diagnosis(self):
             assert ('Category' in line)
             assert ('Defective Details' in line)
             assert ('Index' in line)
+
+    def test_data_diagnosis_run(self):
+        """Test for the run process of rule-based data diagnosis."""
+        test_raw_data = str(Path(__file__).parent.resolve()) + '/test_results.jsonl'
+        test_rule_file = str(Path(__file__).parent.resolve()) + '/test_rules.yaml'
+        test_baseline_file = str(Path(__file__).parent.resolve()) + '/test_baseline.json'
+        # Test - output in excel
+        DataDiagnosis().run(
+            test_raw_data, test_rule_file, test_baseline_file, str(Path(__file__).parent.resolve()), 'excel'
```

Comment:
set "self.parent_path = Path(__file__).parent.resolve()" in init if you need to use it many times?

Reply:
ok